nuget/lib/dependabot/nuget/file_updater.rb

```diff
@@ -62,9 +66,12 @@ def try_update_projects(dependency)
 
           next unless project_dependencies.any? { |dep| dep.name.casecmp(dependency.name).zero? }
 
-          NativeHelpers.run_nuget_updater_tool(repo_root: repo_contents_path, proj_path: proj_path,
-                                               dependency: dependency, is_transitive: !dependency.top_level?,
-                                               credentials: credentials)
+          call_nuget_updater_tool(dependency, proj_path) unless checked_files.include?(project_file.name)
+
+          checked_files.add(project_file.name)
```

Comment:
This key will need to include at least the dependency name being updated. `try_update_projects` will run once for each dependency in `dependencies`. 

It would be worth looking at updating the NuGetUpdater to take in an array of dependency updates to avoid the cost of evaluating the build files multiple times.

Reply:
Good point about the dependency, I think it's also technically the version that needs to be checked as well. I've updated the key.

I think you're right, reducing the number of build evaluations is definitely our biggest perf savings available. I filed https://dev.azure.com/mseng/1ES/_workitems/edit/2145689 to follow up on that.